src/status_im/mailserver/core.cljs

```diff
@@ -523,17 +542,40 @@
                    :mailserver/pending-requests))})
 
 (fx/defn handle-request-completed
```

Comment:
Could you please adding some tests specifying the behavior of this `fn` , as now there are quite a few conditional branches. Thanks!

Reply:
most certainly

Reply:
done. But I didn't know how exhaustive to make the tests. I think I've covered the essentials, but being that this my first time writing tests, I may be doing it all wrong or something.